frontend/drivers/platform_psp.c

```diff
@@ -342,25 +342,30 @@ static void frontend_psp_exec(const char *path, bool should_load_game)
    RARCH_LOG("Attempt to load executable: [%s].\n", path);
 #if defined(VITA)
    RARCH_LOG("Attempt to load executable: %d [%s].\n", args, argp);
+   int ret;
 #ifdef IS_SALAMANDER
    sceAppMgrGetAppParam(boot_params);
    if (strstr(boot_params,"psgm:play"))
    {
-      int ret;
-      char *param1 = strstr(boot_params, "&param=")+7;
+      char *param1 = strstr(boot_params, "&param=");
       char *param2 = strstr(boot_params, "&param2=");
-      memcpy(core_name, param1, param2 - param1);
-      core_name[param2-param1] = 0;
-      sprintf(argp, param2 + 8);
-      ret = sceAppMgrLoadExec(core_name, (char * const*)((const char*[]){argp, 0}), NULL);
-      RARCH_LOG("Attempt to load executable: [%d].\n", ret);
+      if (param1 != NULL && param2 != NULL)
+      {
+         param1 += 7;
+         memcpy(core_name, param1, param2 - param1);
```

Comment:
This is terrible code and can crash easily depending on where param1 and param2 end up in memory. Please fix.

Reply:
Yeah, this is concerning. @crystalct Can you fix this?

Reply:
Not having a development environment for the PSVita and a PSVita, too, I can try to make a comparison with the PS3.